bpf/tests/prepend_name_test.c

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Cilium */
+
+//go:build ignore
+
+#include "vmlinux.h"
+
+#include "bpf_tracing.h" // bpf_printk
+
+#include "bpf_task.h"
+#include "process/bpf_process_event.h"
+
+char _license[] __attribute__((section("license"), used)) = "Dual BSD/GPL";
+
+#define TEST_MAX_BUF_LEN 256
+#define NAME_MAX	 255
+
+struct test_prepend_name_state_map_value {
+	char buf[TEST_MAX_BUF_LEN];
+	u64 buflen;
+	char dname[NAME_MAX];
+	char pad;
+	u32 dlen;
+	u32 offset;
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, int);
+	__type(value, struct test_prepend_name_state_map_value);
+} test_prepend_name_state_map SEC(".maps");
+
+__attribute__((section("raw_tracepoint/test"), used)) int
```

Comment:
This BPF unit testing is extremely valuable and honestly long overdue. Thanks for adding the first of hopefully many to come